### ISSUE TITLE
unable to reset dropzone

### ISSUE THREAD

**Issue by dgraceffa** [peterfreeman/ngx-dropzone#23]
Hi,
I'm trying to reset dropzone's data with dropzone.reset(), but i don't understand if i have to declare something in my component.ts or something else. Maybe there is something i forgot from docs because it throws can't read property reset() of undefined.
Thanks in regard

**Comment by peterfreeman** [comment]
Hi there,

please see the readme documentation. You have to declare a template reference on the dropzone component:

```html
<ngx-dropzone #templRefHere></ngx-dropzone>
<button (click)="templRefHere.reset()">Reset</button>
```

**Comment by dgraceffa** [comment]
Hi there,

thanks for replying, but i wasn't  able to solve the problem.
Take a look at this code: 
` <ngx-dropzone *ngIf="permissions.assets.create && permissions.folders.create" class="custom-dropzone"
      [label]="'Drop your files here to upload '" (filesAdded)="onFilesAdded($event)" [showPreviews]="true" #dropzone>
    </ngx-dropzone>
      <button mat-icon-button (click)="dropzone.reset()">
        <mat-icon>replay</mat-icon>
      </button>`

Isn't this correct?

Thanks

**Comment by peterfreeman** [comment]
Hi,

I've tested your code using the Chrome browser and for me everything is working fine.
Which browser are you using and are you getting any errors in the console?

Happy coding! 🤓

**Comment by dgraceffa** [comment]
Hi,

in console it shows 'can't read property reset of undefined'. Maybe you wrote something in your ts?
I'm using Chrome  too.

thanks

**Comment by peterfreeman** [comment]
Hi,

the reset() method is just a public method of the component and should be accessible:

```js
// in ngx-dropzone.component.ts
reset() {
  this.service.reset();
}
```

Maybe it has something to do with your `ngIf`? So the component element is not available and `#dropzone` is undefined?

**Comment by FelixEhuan** [comment]
If you use` ngIf ` directive, you can pass the reference to the method reset

HTML: 
`<button (click)="reset(dropzone);" *ngIf="file">
  Eliminar foto
</button>
      <ngx-dropzone
        #dropzone
      ></ngx-dropzone>`

TS: 
`  reset(ref) {
    ref.reset();
  }
`

**Comment by peterfreeman** [comment]
> If you use `ngIf` directive, you can pass the reference to the method reset
> 
> HTML:
> `<button (click)="reset(dropzone);" *ngIf="file"> Eliminar foto </button> <ngx-dropzone #dropzone ></ngx-dropzone>`
> 
> TS:
> `reset(ref) { ref.reset(); }`

Yes, that would be a possible workaround. 
Otherwise you could also wrap the dropzone and the button in a `div` and apply the `*ngIf` to that.
Template references don't work with `*ngIf`: see [StackOverflow](https://stackoverflow.com/questions/44811132/how-do-i-combine-a-template-reference-variable-with-ngif).

**Comment by SalkiDixit** [comment]
@peterfreeman  hi I am also facing same issue, not able to reset dropzone, i am using inside my angular7 project, i tried what u have suggested above, but still stucked, please asist.

**Comment by peterfreeman** [comment]
Please post some code, otherwise I cannot help you.

**Comment by SalkiDixit** [comment]
@peterfreeman its done but now i am facing other issue, while previewing files, video is not visible accuratly, i can see name of video and it is uploading properly but preview is not looking well.
Would you please guide me here.

**Comment by peterfreeman** [comment]
Currently, there is only a preview support for images. Videos and all other files get displayed with a light gray background. I might implement video support in the next version I'm currently working on.

**Comment by Ebsan** [comment]
Was this feature taken out? It looks like reset was removed around this pull because file cacheing is now up to the parent component? (https://github.com/peterfreeman/ngx-dropzone/pull/29)

I was able to hack a way around it though by manually filtering the _previewChildren:

```
@ViewChild(NgxDropzoneComponent)
dz!: NgxDropzoneComponent;

...

this.dz._previewChildren.filter(() => false);
```

I'm still new to Angular and TS so this might not be the best way to do it but it looks like it works.
